### PR TITLE
B::Deparse: Handle try/catch when catch has an ENTER/LEAVE pair

### LINKED ISSUE #18874: Deparsing of try/catch cannot handle ENTER/LEAVE wrapped catch
B::Deparse does not know how to handle catch{} if it has a full ENTER/LEAVE scope.

This works:
```
$ ./perl -Ilib -MO=Deparse -e 'use feature qw(try); sub foo { try { 1; } catch ($e) { $e; } }'
try/catch is experimental at -e line 1.
try/catch is experimental at -e line 1.
use feature 'try';
sub foo {
    try {
        1;
    }
    catch($e) {
        $e;
    }
}
-e syntax OK
```
This doesn't:
```
$ ./perl -Ilib -MO=Deparse -e 'use feature qw(try); sub foo { try { 1; } catch ($e) { my $x; $e; } }'
try/catch is experimental at -e line 1.
try/catch is experimental at -e line 1.
While deparsing -e near line 1,
Expected scope as second child of catch at lib/B/Deparse.pm line 4075.
CHECK failed--call queue aborted.

```

### PR DESCRIPTION
**Pull request by wolfsage**
Fixes #18874

### CODE REVIEW

File: lib/B/Deparse.pm [hunk 1]
```diff
@@ -4072,11 +4072,15 @@ sub pp_leavetrycatch {
     $catch->name eq "catch" or die "Expected catch as third child of leavetrycatch";
 
     my $catchblock = $catch->first->sibling;
-    $catchblock->name eq "scope" or die "Expected scope as second child of catch";
+    my $name = $catchblock->name;
+    unless ($name eq "scope" || $name eq "leave") {
+      "Expected scope or leave as second child of catch, got $name instead\n";
```
Comment (leonerd) [inline lib/B/Deparse.pm]: Did you mean to maybe "return" or "die" this string?

Comment (wolfsage) [reply lib/B/Deparse.pm]: Bah, fixed

**Comment by leonerd** [comment]
LGTM

**Review by leonerd — APPROVED** [review]
LGTM

**Comment by wolfsage** [comment]
Thanks, merged

**Pull request closed without merging**